LibreNMS/OS/Iosxe.php

```diff
@@ -43,10 +54,103 @@ class Iosxe extends Ciscowlc implements
+        'l1IntermediateSystem' => 'L1',
+        'l2IntermediateSystem' => 'L2',
+        'l1L2IntermediateSystem' => 'L1L2',
+        'unknown' => 'unknown',
+    ];
+
+    public function discoverIsIs(): Collection
+    {
+        // Check if the device has any ISIS enabled interfaces
+        $circuits = snmpwalk_cache_oid($this->getDeviceArray(), 'CISCO-IETF-ISIS-MIB::ciiCirc', []);
+        $adjacencies = new Collection;
+
+        if (! empty($circuits)) {
+            $adjacencies_data = snmpwalk_cache_twopart_oid($this->getDeviceArray(), 'CISCO-IETF-ISIS-MIB::ciiISAdj', [], null, null, '-OQUstx');
+            $ifIndex_port_id_map = $this->getDevice()->ports()->pluck('port_id', 'ifIndex');
+
+            // No ISIS enabled interfaces -> delete the component
+            foreach ($circuits as $circuit_id => $circuit_data) {
+                if (! isset($circuit_data['ciiCircIfIndex'])) {
+                    continue;
+                }
+
+                if ($circuit_data['ciiCircPassiveCircuit'] == 'true') {
+                    continue; // Do not poll passive interfaces
+                }
+
+                $adjacency_data = Arr::last($adjacencies_data[$circuit_id] ?? [[]]);
+
+                $adjacency = new IsisAdjacency([
+                    'device_id' => $this->getDeviceId(),
+                    'ifIndex' => $circuit_data['ciiCircIfIndex'],
+                    'port_id' => $this->ifIndexToId($circuit_data['ciiCircIfIndex']),
+                    'isisCircAdminState' => $circuit_data['ciiCircAdminState'] ?? 'down',
+                    'isisISAdjState' => $adjacency_data['ciiISAdjState'] ?? 'down',
+                ]);
+
+                if (! empty($adjacency_data)) {
+                    $adjacency->isisISAdjNeighSysType = Arr::get($this->isis_codes, $adjacency_data['ciiISAdjNeighSysType'] ?? 'unknown', 'unknown');
+                    $adjacency->isisISAdjNeighSysID = str_replace(' ', '.', trim($adjacency_data['ciiISAdjNeighSysID'] ?? ''));
+                    $adjacency->isisISAdjNeighPriority = $adjacency_data['ciiISAdjNeighPriority'] ?? '';
+                    $adjacency->isisISAdjLastUpTime = $this->parseAdjacencyTime($adjacency_data);
+                    $adjacency->isisISAdjAreaAddress = str_replace(' ', '.', trim($adjacency_data['ciiISAdjAreaAddress'] ?? ''));
+                    $adjacency->isisISAdjIPAddrType = $adjacency_data['ciiISAdjIPAddrType'] ?? '';
+                    $adjacency->isisISAdjIPAddrAddress = (string) IP::fromHexstring($adjacency_data['ciiISAdjIPAddrAddress'] ?? null, true);
+                }
+
+                $adjacencies->push($adjacency);
+            }
+        }
+
+        return $adjacencies;
+    }
+
+    public function pollIsIs($adjacencies): Collection
+    {
+        $data = snmpwalk_cache_twopart_oid($this->getDeviceArray(), 'ciiISAdjState', [], 'CISCO-IETF-ISIS-MIB');
+
+        if (count($data) !== $adjacencies->where('ciiISAdjState', 'up')->count()) {
```

Comment:
Are you sure this is right?

The walk should contain all adjacencies... but you are only counting ones that are up (and excluding passive ones).

Seems like we could run discovery every poll on a real instance.